peps/pep-0735.rst

```diff
@@ -0,0 +1,613 @@
+Specifiers or else PEP 735 Dependencies.
+
+PEP 735 Dependencies
+--------------------
+
+PEP 735 Dependencies are objects which define a set of dependencies. They do
+not necessarily refer to a single package, as will become clear from the
+definitions below.
+
+There are two forms of PEP 735 Dependencies: "Dependency Group Includes" and
+"Path Dependencies".
+
+Dependency Group Include
+''''''''''''''''''''''''
+
+A Dependency Group Include includes the dependencies of another Dependency
+Group in the current Dependency Group.
+
+An include is defined as an object with exactly one key, ``"include"``, whose
+value is a string, the name of another Dependency Group.
+
+For example, ``{include = "test"}`` is an include which expands to the
+contents of the ``test`` Dependency Group.
+
+Path Dependency
+'''''''''''''''
+
+A Path Dependency is a dependency on a package found via a local filesystem
+path OR on dependencies of a package found via a local filesystem path.
+
+It contains the following keys, with associated types:
+
+* ``path``: a string, required
+* ``extras``: a list of strings, optional
+* ``editable``: a boolean, optional
+* ``only_deps``: a boolean, optional
+
+For a simple example, ``{path = ".", editable = true, extras = ["mysql"]}`` is
+a Path Dependency on the current project including its ``mysql`` extra. In
+``requirements.txt`` files, a similar idea can be expressed as ``-e .[mysql]``.
+
+``path``
+~~~~~~~~
+
+The ``path`` must refer to the path to a built distribution (wheel, sdist, or
+any future file format) or a directory containing python package source code.
+
+If the path refers to a built distribution, that package should be installed
+when the Dependency Group is installed.
+
+If the path refers to a directory, the directory SHOULD be a valid Python
+package. Implementations MAY refuse to process ``path`` directives which are
+not packages, even when ``only_deps`` is specified (see below for how
+``only_deps`` would potentially make it possible to process non-package paths).
+
+``extras``
+~~~~~~~~~~
+
+The ``extras`` key is a list of strings, each of which is the name of an extra
+which should be included in the package installation.
+
+``editable``
+~~~~~~~~~~~~
+
+If ``editable`` is ``true``, implementations installing the dependency group
+SHOULD install the dependency in editable mode. If it is ``false``, they SHOULD
+install it in non-editable mode. If ``editable`` is absent, no default behavior
+is specified and implementations may choose their preferred default.
+
+Implementations MAY provide options to users to configure or override this behavior.
+For example, a tool may have an option ``--never-editable`` which always treats
+``editable`` as ``false``.
+However, implementations SHOULD prefer to use the ``editable`` value if it is
+present.
+
+If ``editable`` is specified on a Path Dependency which refers to a built
+distribution, implementations MUST treat this as an error.
+
+``only_deps``
```

Comment:
Nit: I think a hyphen (`only-deps`) is more consistent with other parts of pyproject.toml, e.g. this PEP's own `dependency-groups` table.

Reply:
Done! (Although, aside: based on the discussion that field might be going away anyway. It seems I misunderstood the needs which prompted it.)